app/features/build_runner/fastlane_build_runner.rb

```diff
@@ -128,6 +135,7 @@ def run(new_line_block:, completion_block:)
                 end.join("\n"))
               end
 
+              # Install the new Bundle and require all the new gems into the runtime.
```

Comment:
How do we handle multiple builds here? I'm concerned that we might get into a state where we inject dependencies at the same time as another build runner is removing them. How can we be sure that we don't get into this state?

Reply:
AFAIK, being in the `Bundler::SharedHelpers.chdir` block, the process is synchronized, therefore this _shouldn't_ be an issue. But this will need a lot of testing tho.

Reply:
Yeah but we're talking about 3 parts here: `saving state, modifying it` -> `running build` -> `restoring state`
If this process isn't atomic throughout, then another build running might interlace operations

Reply:
Two options:
1. Synchronize it (CI's `Bundle` is atomic, and then builds are too).
2. Do not synchronize it, erase the restoration process, just get along with the injected deps until we can really isolate the process.